### DISCUSSION TITLE
Form of prop_err and add_err from nlmixr?

### DISCUSSION THREAD

**Discussion by Dstix** [nlmixr2/rxode2#371] [Q&A] [answered]
Is the output of prop_err or add_err in variance or standard deviation? In previous versions of nlmixr (nlmixr version 2.0.7) you could type `fit$sigma` and get the same prop_err value. In nlmixr2 version 2.0.8 the only place this appears is under `fit$fixef[XX]`. Is this correct? The reason I ask is I used the `fit$sigma` as input into my mrgsolve simulations and I want to make certain I correctly input variance or the standard deviation.

**Comment by mattfidler** [comment, marked as answer]
Hi @Dstix;  The `prop_err` and `add_err` represents the standard deviation.

You could also directly simulate from `rxode2` using the fit, this uses a `fit$sigma` as an identity matrix and uses the `fixef[XX]`  as a scaling parameter for these values.

**Reply by billdenney** [reply]
FYI, to simulate, you can use your model fit object and change `est = "rxSolve"`, like:

```r
nlmixr(fit, data=simdata, est = "rxSolve")
```

**Reply by Dstix** [reply]
Thanks. I am not as conversant with rxODE having relied on mrgsolve for so long. Is this the same for fit$omega? Are the etas in the omega block in variance or standard deviation? I have been using these blocks in mrgsolve in the following form (where mod is the mrgsolve model):

`mod %<>% omat(dmat(
  fit$omega[1], # CL
  fit$omega[4],  # VC
  0,             # Q
  0              # VP
))

mod %<>% smat(dmat(fit$sigma))'

**Reply by Dstix** [reply]
Based on this site (https://rdrr.io/github/nlmixr2/rxode2/man/ini.html), I am taking the `fit$omega` to be the variance-covariance:

> 'omega' values can be set as a single value or as the values of a lower-triangular matrix. The values may be set as either a variance-covariance matrix (the default) or as a correlation matrix for the off-diagonals with the standard deviations on the diagonals.

**Reply by mattfidler** [reply]
Yes the `fit$omega` is the variance-covariance matrix.

The `fit$omegaR` is the correlation matrix with the diagonals being the standard deviation of each component.

**Reply by mattfidler** [reply]
(I believe it is printed out in the fit object in case you want a quick place to figure it out)

**Reply by Dstix** [reply]
Ahh - I see it now:

```
 Full BSV covariance (fit$omega) or correlation (fit$omegaR; diagonals=SDs) 
```

This really helps. Thanks!

**Reply by mattfidler** [reply]
> Thanks. I am not as conversant with rxODE having relied on mrgsolve for so long. Is this the same for fit$omega? Are the etas in the omega block in variance or standard deviation? I have been using these blocks in mrgsolve in the following form (where mod is the mrgsolve model):
> 
> `mod %<>% omat(dmat( fit$omega[1], # CL fit$omega[4], # VC 0, # Q 0 # VP ))
> 
> mod %<>% smat(dmat(fit$sigma))'

It seems OK (though I am not an expert in `mrgsolve`);  The `$sigma` is fixed to one in `nlmixr2` estimation for `focei`, so I believe you should still do the same thing in `mrgsolve` if you use this method.

Also if you use `mrgsolve` keep in mind that they use `NOCB` for covariates, whereas `nlmixr2` uses `LOCF` for time-varying covarites.  I think you can use `LOCF` with `mrgsolve` but I cannot remember how to specify it off-hand.

On the other hand the `rxode2` model can now simply be the UI function you use in `nlmixr2`. (you can extract it from the fit by using `as.function(fit)`, though it is easier to simulate from the fit directly since it will grab the right `omega` matrix for you directly (and specify the degrees of freedom needed for simulation with uncertainty if you take this approach)

**Reply by Dstix** [reply]
Yes, moving forward I will be using the rxSolve function. Much handier working with nlmixr models.
